src/uu/cksum/src/cksum.rs

```diff
@@ -440,6 +566,34 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
         .into());
     }
 
+    if check {
```

Comment:
If this branch is taken, then the value of `--length` is never even sanity-checked:

```console
$ echo 'e5773576fc75ff0f8eba14f61587ae28  README.md' > foo.sums 
$ ../gnu/src/cksum -a md5 -l 65536 -c foo.sums 
../gnu/src/cksum: --length is only supported with --algorithm=blake2b
[$? = 1]
$ cargo run -q cksum -a md5 -l 65536 -c foo.sums 
README.md: OK
```

Note that this is a distinct issue from the "`MD5-65536` algorithm-in-sums-file" issue.